### ISSUE TITLE
For polar and variwide charts, initial animation is restarted if reflow occurs while it is running

### ISSUE THREAD

**Issue by piet23** [highcharts/highcharts#19443]
#### Expected behaviour
Initial animation to be run only once, for any type of chart.

#### Actual behaviour
When a reflow occurs (e.g., because a scrollbar appears and thus resizes the chart's width) while the initial animation is running, the animation is restarted. This happens at least for polar charts and for charts with type variwide, but not for normal spline type charts.

#### Live demo with steps to reproduce
https://jsfiddle.net/piet23/La95cm40/2/
The window with the rendered charts should be large enough that initially all three charts fit in. Their initial animation is set to run for 1s. After 0.5s, a huge amount of text is added below the charts, resulting in a sidebar to appear, which resizes the charts and then restarts the animation of the polar and variwide charts.

#### Product version
Highcharts 11.1.0

When loading Highcharts 9.0 or 10.0 libraries, the behaviour is as expected.

#### Affected browser(s)
Firefox, Chrome, Edge

**Comment by karolkolodziej** [comment]
Hi @piet23!

Thank you for sharing this issue! However, in my case, the animation isn't interrupted by any reflow.
Am I missing something here? I've tried different window sizes but they are still the same.

> When loading Highcharts 9.0 or 10.0 libraries, the behaviour is as expected.

In version 11 we've changed how the `reflow` event is attached (`ResizeObserver` was added) but since I'm unable to recreate that it is hard to tell what might be wrong.

**Comment by piet23** [comment]
Hi @karolkolodziej,

thanks for the reply. I have now noticed that this issue happens in Windows, but not in MacOS (at least the version I could test). Looks like in MacOS the scrollbars behave differently and do not take up space, and therefore do not resize the charts. Maybe that can explain why you cannot reproduce it?

I have reduced the fiddle a bit, keeping only the variwide chart to make the displayed stuff smaller, so that it should be easier to reproduce, without having to rearrange the windows. That version is here: https://jsfiddle.net/piet23/La95cm40/8/

As I noted originally, the issue occurs if at the start there is no scrollbar but during the initial animation a scrollbar appears and resizes the chart.

**Comment by karolkolodziej** [comment]
You are right I'm using a Mac which is maybe why I'm not able to see this.
I've also tried your second demo but with no luck. Let me organize a Windows machine where I can test that.

**Comment by KacperMadej** [comment]
The problem is visible on Windows because the horizontal scrollbar take inner space by default forcing a reflow that interrupts the initial animation.
The below minimal demo should work the same on any OS:
https://jsfiddle.net/BlackLabel/mnja28zw/1/

The initial animation is interrupted via setSize and forces restart while the total timer is not changed causing the animation to restart a few times and jump to the end after the animation time.

What is the expected behavior on interruption of the animation? Continue, finish, or something else?

A bonus question that might help with prioritization of this issue:
Is there a real case scenario when the initial animation is interrupted purposefully and not due to a design mishap?

**Comment by piet23** [comment]
I would expect the drawing to continue, like it does when using a simple spline chart.

We don't purposefully interrupt the initial animation, but have several places where the charts and other data is loaded from the backend and the mentioned behavior of the sidebar appearing during the initial animation happens.

**Comment by karolkolodziej** [comment]
I'm able to recreate this issue on macOS.
